src/extensions/default/bramble-move-file/MoveUtils.js

```diff
@@ -0,0 +1,59 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true,  regexp: true, indent: 4, maxerr: 50 */
+/*global define, brackets, $ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var BracketsFiler   = brackets.getModule("filesystem/impls/filer/BracketsFiler");
+    var BlobUtils       = brackets.getModule("filesystem/impls/filer/BlobUtils");
+    var Path            = BracketsFiler.Path;
+
+    var NEEDS_RENAME    = "needs rename";
+
+    function move(src, dest) {
+        var deferred = new $.Deferred();
+        var fs = BracketsFiler.fs();
+        src = src.replace("/?$", "");
+        dest = dest.replace("/?$", "");
+        var srcNodeName = Path.basename(src);
+
+        if(src === dest || Path.dirname(src) === dest) {
+            return deferred.resolve();
+        }
+
+        fs.readdir(dest, function(err, contents) {
+            if(err) {
+                return deferred.reject(err);
+            }
+
+            if(contents.indexOf(srcNodeName) !== -1) {
+                return deferred.reject({ type: NEEDS_RENAME });
+            }
+
+            fs.mv(src, dest, function(err) {
+                if(err) {
+                    return deferred.reject(err);
+                }
+
+                BlobUtils.preload(dest, function(err) {
```

Comment:
Do this after you're done changing the fs (i.e., after the `rm` below) as the last thing you do before you resolve.